File: ast/ast_types/builtin.go
package ast_types

import (
	"github.com/panda-io/micro-panda/ast/ast"
	"github.com/panda-io/micro-panda/token"
)

type TypeBuiltin struct {
	TypeBase
	Token token.Token
}

func (b *TypeBuiltin) Equal(t ast.Type) bool {
	if t, ok := t.(*TypeBuiltin); ok {
		return t.Token == b.Token
	}
	return false
}

func (b *TypeBuiltin) String() string {
	return b.Token.String()
}

File: MicroPanda/AST/Type/Builtin.cs
namespace MicroPanda.AST.Type;

using Token;

internal class Builtin : Type
{
    internal Token Token { get; set; }

    internal Builtin(Token token)
    {
        Token = token;
    }
    
    internal override bool Equal(Type type)
    {
        if (type is Builtin typeBuiltin)
        {
            return typeBuiltin.Token == Token;
        }
        return false;
    }
}